### ISSUE TITLE
Pre-cache songs in queue

### ISSUE THREAD

**Issue by el-tiuri** [BLeeEZ/amperfy#152]
It would be nice to pre-cache the next one or two songs in the play queue so it doesn’t start loading when the song is playing. This would reduce pauses between songs, especially on spotty connections.

**Comment by BLeeEZ** [comment]
This is already implemented.
Enable the option:  Settings -> Player -> "Auto Cache Played Songs"
With the option enabled the next 3 upcoming songs/podcast episodes will be pre-downloaded.
I think i should add a hint to make this pre-download functional visible to the user.

**Comment by el-tiuri** [comment]
But doesn’t this mean the songs remain cached, thus filling up space? That’s how I understood it, at least.

**Comment by BLeeEZ** [comment]
Yes the songs are cached permanently. For caching currently the download api function is needed. The behaviour and user restrictions for this api function are different for each server. Because of this I will not add an other type of caching for user which have caching disabled. Depending on the backend is behaviour would lead to difficult error messages.

**Comment by mneuburg-livefront** [comment]
I assure you that this is just not true. Amperfy is completely incapable of gapless playback, no matter how you twiddle with the settings. For classical music where continuous sound is broken up into smaller tracks, it's just plain useless
